temporal-sdk/src/main/java/io/temporal/worker/tuning/JVMSystemResourceInfo.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import com.sun.management.OperatingSystemMXBean;
```

Comment:
This might pose a problem for some module-based users. I think they may have to "add opens" and such to their module setups. We may need a Java SDK sample or at least confirmation that module-based usage of the Java SDK does not have issues here. Worst case scenario, resource-based tuning could move to a separate JAR.

Reply:
Good point. I believe this should be accessible since the javadocs include it: https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html and you'd think that means it's part of the open API, but, I can try to confirm that.

Reply:
I have found using JVM modules in modern Java versions that if you access `com.sun` things these days it poses a big problem and is a pain (you have to "add opens" and all this other mess).